client-v2/src/shared/actions/ArticleFragments.ts

```diff
@@ -112,23 +115,62 @@ const replaceGroupArticleFragments = (
   }
 });
 
-function addArticleFragment(id: string) {
-  return (dispatch: Dispatch) =>
-    getArticles([id])
-      .then(([article]) => {
-        dispatch(externalArticleActions.fetchSuccess([article]));
-        const fragment = createFragment(article.id);
-        dispatch(
-          articleFragmentsReceived({
-            [fragment.uuid]: fragment
-          })
-        );
-        return fragment;
+/**
+ * Given a resource id, create an article fragment and add it to the store.
+ * The resource id can be a few different things:
+ *  - a article, tag or section (either the full URL or the ID)
+ *  - an external link.
+ */
+function createArticleFragment(
+  resourceId: string
+): ThunkResult<Promise<ArticleFragment | undefined>> {
+  return async (dispatch: Dispatch) => {
+    const isURL = isValidURL(resourceId);
+    const id = isURL ? getIdFromURL(resourceId) : resourceId;
+    let fragment: ArticleFragment;
+    if (isURL && !id) {
+      // If we have a URL from an external site,
+      // create a snap of type 'link'.
+      fragment = await createLinkSnap(resourceId);
```

Comment:
Are we not returning anything in this instance? Only asking because I am interested in getting rid of the elses!

Reply:
... yeah, I've refactored this to be a bit less smelly.